File: software/examples/rust/example_indicator.rs
use std::{error::Error, io};

use tinkerforge::{ip_connection::IpConnection, motion_detector_v2_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Motion Detector Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let md = MotionDetectorV2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Turn blue backlight LEDs on (maximum brightness)
    md.set_indicator(255, 255, 255);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_indicator.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/motion_detector_v2_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Motion Detector Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	md, _ := motion_detector_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Turn blue backlight LEDs on (maximum brightness)
	md.SetIndicator(255, 255, 255)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
